frontend/src/stores/theme.ts

```diff
@@ -0,0 +1,28 @@
+import { writable } from 'svelte/store';
+
+function getInitialTheme() {
+  const savedTheme = localStorage.getItem('theme');
+  if (savedTheme !== null) {
+    return savedTheme;
+  }
+  return window.matchMedia('(prefers-color-scheme: dark)').matches ? 'dark' : 'light';
+}
+
+export const theme = writable<string>(getInitialTheme());
+
+// Subscribe to theme changes
+theme.subscribe((value) => {
```

Comment:
Since this is startup code that attaches subscribers, it should not be on the top level but wrapped in a function that is called from `main.ts`